### PR TITLE
sl-155: The password can be changed with the same value as the current password

### PR DESCRIPTION
**Pull request by Izdormu**
errormessage

### CODE REVIEW

File: frontend/src/libs/components/error-message/error-message.tsx [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import { getValidClassNames } from '~/libs/helpers/helpers.js';
+
+import styles from './styles.module.scss';
+
+type Properties = {
+  error?: string;
```
Comment (andanf-e) [inline frontend/src/libs/components/error-message/error-message.tsx]: Or `null` - cause actually we'd always pass this prop as a value from `errors` obj
```suggestion
  error: string | undefined;
```

File: frontend/src/libs/components/input/input.tsx [hunk 2]
```diff
@@ -70,9 +71,10 @@ const Input = <T extends FieldValues>({
         disabled={isDisabled}
         inputMode={inputMode}
       />
-      <span className={styles['error-message']}>
-        {hasError && (error as string)}
-      </span>
+      <ErrorMessage
+        error={error as string}
+        className={styles['error-message']}
```
Comment (andanf-e) [inline frontend/src/libs/components/input/input.tsx]: If this custom className brings only a `min-height` for container - won't it be better to put it inside `ErrorMessage` styles?

File: frontend/src/libs/components/select/select.tsx [hunk 3]
```diff
@@ -77,9 +77,10 @@ const Select = <T extends FieldValues>({
         name={name}
         styles={styles}
       />
-      <span className={style['error-message']}>
-        {hasError && (error as string)}
-      </span>
+      <ErrorMessage
+        error={error as string}
+        className={style['error-message']}
```
Comment (andanf-e) [inline frontend/src/libs/components/select/select.tsx]: Same about custom className for `min-height`

File: frontend/src/libs/components/input/input.tsx [hunk 4]
```diff
@@ -70,9 +71,7 @@ const Input = <T extends FieldValues>({
         disabled={isDisabled}
         inputMode={inputMode}
       />
-      <span className={styles['error-message']}>
-        {hasError && (error as string)}
-      </span>
+      <ErrorMessage error={error as string} />
```
Comment (andanf-e) [inline frontend/src/libs/components/input/input.tsx:73]: Have you removed in `styles` for `input` this `error-message` styling?

Comment (what1s1ove) [reply frontend/src/libs/components/input/input.tsx:73]: agreed

**Review by what1s1ove — CHANGES_REQUESTED** [review]

File: frontend/src/libs/components/input/input.tsx [hunk 5]
```diff
@@ -10,6 +10,7 @@ import { getValidClassNames } from '~/libs/helpers/helpers.js';
 import { useFormController } from '~/libs/hooks/hooks.js';
 import { type IconType } from '~/libs/types/types.js';
 
+import { ErrorMessage } from '../error-message/error-message.js';
```
Comment (Alyona-Pisotska) [inline frontend/src/libs/components/input/input.tsx]: fix import `'~/libs/components/components.js'`

File: frontend/src/libs/components/select/select.tsx [hunk 6]
```diff
@@ -9,7 +9,7 @@ import ReactSelect from 'react-select';
 import { useCallback, useFormController } from '~/libs/hooks/hooks.js';
 import { type SelectOption } from '~/libs/types/types.js';
 
-import style from './select.module.scss';
+import { ErrorMessage } from '../error-message/error-message.js';
```
Comment (Alyona-Pisotska) [inline frontend/src/libs/components/select/select.tsx]: change import

**Review by Alyona-Pisotska — CHANGES_REQUESTED** [review]

File: frontend/src/libs/components/input/input.tsx [hunk 7]
```diff
@@ -70,9 +70,7 @@ const Input = <T extends FieldValues>({
         disabled={isDisabled}
         inputMode={inputMode}
       />
-      <span className={styles['error-message']}>
-        {hasError && (error as string)}
-      </span>
+      <ErrorMessage error={error as string} />
```
Comment (andanf-e) [inline frontend/src/libs/components/input/input.tsx:73]: I think you have no need to cast it to string

File: frontend/src/libs/components/select/select.tsx [hunk 8]
```diff
@@ -108,9 +107,7 @@ const Select = <T extends FieldValues>({
         onMenuClose={handleMenuOpenToggle}
         styles={getStyles({ hasIcon: !isMenuOpen && !isOptionSelected })}
       />
-      <span className={styles['error-message']}>
-        {hasError && (error as string)}
-      </span>
+      <ErrorMessage error={error as string} />
```
Comment (andanf-e) [inline frontend/src/libs/components/select/select.tsx:110]: Same about type casting

File: frontend/src/libs/components/input/input.tsx (hunk 7 above)
Comment (Izdormu) [reply frontend/src/libs/components/input/input.tsx:73]: if i not add this, i will have error
Type 'string | ("message" extends keyof FieldError & keyof DeepRequired<T>[string] ? [FieldError["message" & keyof DeepRequired<T>[string]], FieldErrorsImpl<...>["message" & keyof DeepRequired<...>[string]]] extends [...] ? Merge<...> : FieldError["message" & keyof DeepRequired<...>[string]] | FieldErrorsImpl<...>["messag...' is not assignable to type 'string | undefined'.

File: frontend/src/libs/components/select/select.tsx (hunk 8 above)
Comment (Izdormu) [reply frontend/src/libs/components/select/select.tsx:110]: (property) error: string | ("message" extends keyof FieldError & keyof DeepRequired<T>[string] ? [FieldError["message" & keyof DeepRequired<T>[string]], FieldErrorsImpl<...>["message" & keyof DeepRequired<...>[string]]] extends [...] ? Merge<...> : FieldError["message" & keyof DeepRequired<...>[string]] | FieldErrorsImpl<...>["message" & keyof DeepRequired<...>[string]] : string | undefined) | undefined

**Review by andanf-e — APPROVED** [review]

**Review by what1s1ove — APPROVED** [review]

**Pull request merged**